steps.go

```diff
@@ -576,21 +578,22 @@ func (tc *TestContext) getMonitoringServerIp() (string, error) {
 		}
 		for ip := range ips {
 			if _, ok := instancePoolIps[ip]; !ok {
-				if foundIp != "" {
-					return "", fmt.Errorf("too many non-instancepool VM's running: %d", len(ips))
-				}
-				foundIp = ip
+				nonPoolIps = append(nonPoolIps, ip)
 			}
 		}
 	} else {
 		if len(ips) != 1 {
 			return "", fmt.Errorf("too many non-instancepool VM's running: %d", len(ips))
 		}
 	}
-	if foundIp == "" {
+	if len(nonPoolIps) > 1 {
+		tc.logger.Debugf("found the following non-instancepool VM IPs: %v", nonPoolIps)
+		return "", fmt.Errorf("too many non-instancepool VM's running: %d", len(nonPoolIps))
+	}
+	if len(nonPoolIps) == 0 {
 		return "", fmt.Errorf("monitoring server not running")
 	}
-	return foundIp, nil
+	return nonPoolIps[0], nil
```

Comment:
### Change
Instead of a single (=the first) ip that was found (`foundIp`), changed to a slice with all found ips (`nonPoopIps`)

### Reason
- Logging of all (non instancepool) IPs that were  found
- Having such an additional slice, made `foundIp` redundant
- (Perhaps) improved readability 
    (`foundIp != ""` within loop vs `len(nonPoolIps) > 1` outside of it)
    (`foundIp == ""` vs `len(nonPoolIps) == 0`, etc.)